File: src/main/java/com/github/asm0dey/m4bconverter/model/Track.java
package com.github.asm0dey.m4bconverter.model;

/**
 * User: finkel
 * Date: 13.10.13
 * Time: 13:09
 */
public class Track {
    private String fileName;
    private String track;

    public Track(String fileName) {
        this.fileName = fileName;
    }

    public Track(String fileName, String track) {
        this.fileName = fileName;
        this.track = track;
    }

    public Track() {
    }

    public String getFileName() {
        return fileName;
    }

    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    public String getTrack() {
        return track;
    }

    public void setTrack(String track) {
        this.track = track;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Track)) return false;

        Track track1 = (Track) o;

        if (fileName != null ? !fileName.equals(track1.fileName) : track1.fileName != null) return false;
        if (track != null ? !track.equals(track1.track) : track1.track != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = fileName != null ? fileName.hashCode() : 0;
        result = 31 * result + (track != null ? track.hashCode() : 0);
        return result;
    }
}

File: src/main/java/com/github/asm0dey/m4bconverter/Track.kt
package com.github.asm0dey.m4bconverter.model

import com.github.asm0dey.m4bconverter.gui.GuiMain
import java.io.File
import org.apache.commons.io.FilenameUtils

/**
 * User: finkel
 * Date: 14.10.13
 * Time: 0:55
 */
public data class Track(var file: File, var track: String, var length: Long) : Comparable<Track>, IKnowMyValues, ICanSetMyValues<Track>{
    override fun compareTo(other: Track) = file.compareTo(other.file)
    override fun setValue(value: Any?, column: Int): Track {
        if (getColumnsAndEditable().map { it.second }[column] == true )
            when (column) {
                1 -> track =  value as String
            }
        return this
    }
    override fun getValue(column: Int): Any? {
        return when (column) {
            0 -> FilenameUtils.getName(file.getAbsolutePath())
            1 -> track
            2 -> GuiMain.getBeautifulTime(length)
            else -> null
        }
    }
    class object {
        public fun getColumnsAndEditable(): List<Pair<String, Boolean>> = listOf("File name" to false, "Track name" to true, "Length" to false)
    }
}
public trait IKnowMyValues{
    fun getValue(column: Int): Any?
}
public trait ICanSetMyValues<T>{
    fun setValue(value: Any?, column: Int): T
}

